### PR TITLE
Allow passing arbitrary expressions as memory view location

### PR DESCRIPTION
**Pull request by simark**
Details in each commit's message.

### CODE REVIEW

**Comment by marechal-p** [comment]
This change will require a CQ for the `long` package.

File: packages/cpp-debug/src/browser/memory-provider.ts [hunk 1]
```diff
@@ -17,9 +17,18 @@
...
+export interface MemoryReadResult {
+    bytes: Uint8Array,
+    address: number,
+}
+
 export const MemoryProvider = Symbol('MemoryProvider');
 export interface MemoryProvider {
-    readMemory(address: number, length: number): Promise<Uint8Array>;
+    /**
+     * Read `number` bytes of memory at address `location`, which can be
```
Comment (marcdumais-work) [inline packages/cpp-debug/src/browser/memory-provider.ts]: update doc

**Comment by simark** [comment]
> This change will require a CQ for the `long` package.

Ahh actually I made a version that doesn't use the `long` package (I thought we could add it later).  I just forgot to remove the typings file.

**Comment by simark** [comment]
> Ahh actually I made a version that doesn't use the `long` package (I thought we could add it later). I just forgot to remove the typings file.

Updated.

File: packages/cpp-debug/package.json [hunk 2]
```diff
@@ -25,6 +25,10 @@
     "@theia/debug": "next"
   },
   "devDependencies": {
+    "@types/chai": "^4.1.7",
+    "@types/mocha": "^5.2.0",
+    "chai": "^4.2.0",
+    "mocha": "^5.2.0",
```
Comment (marechal-p) [inline packages/cpp-debug/package.json]: Might be silly, but these are new dependencies, and I wonder if they too should require a CQ :/

**Pull request merged**